Essentials/src/com/earth2me/essentials/commands/Commandeco.java

```diff
@@ -55,13 +62,15 @@ public void run(final Server server, final CommandSource sender, final String co
     protected void updatePlayer(final Server server, final CommandSource sender, final User player, final String[] args) throws NotEnoughArgumentsException, ChargeException, MaxMoneyException {
         switch (cmd) {
             case GIVE:
-                player.giveMoney(amount, sender, UserBalanceUpdateEvent.Cause.COMMAND_ECO);
+                if (isPercent)
+                    player.percentHandler(amount, sender, UserBalanceUpdateEvent.Cause.COMMAND_ECO, 0);
```

Comment:
Please refactor into something along the lines of 'giveMoneyPercent' instead of using these magic values and a switch.